File: src/java/nz/co/searchwellington/model/DiscoveredFeed.java
package nz.co.searchwellington.model;

import java.util.Set;

public interface DiscoveredFeed {

    public int getId();
    public void setId(int id);

    public String getUrl();
    public void setUrl(String url);
    public Set<Resource> getReferences();
    public void setReferences(Set<Resource> references);


}

File: src/main/scala/nz/co/searchwellington/model/DiscoveredFeed.scala
package nz.co.searchwellington.model

import reactivemongo.api.bson.BSONObjectID

import java.util
import java.util.Date
import scala.jdk.CollectionConverters._

case class DiscoveredFeed(_id: BSONObjectID = BSONObjectID.generate, url: String, occurrences: Seq[DiscoveredFeedOccurrence], firstSeen: Date, publisher: Option[BSONObjectID]) {
  def getUrl: String = url
  def getOccurrences: util.List[DiscoveredFeedOccurrence] = occurrences.asJava
  def getFirstSeen: Date = occurrences.headOption.map(_.seen).orNull  // TODO confirm ordering
}

case class DiscoveredFeedOccurrence(referencedFrom: String, seen: Date) {
  def getReferencedFrom: String = referencedFrom
  def getSeen: Date = seen
}
